### ISSUE TITLE
gplenforced.org code hosting currently fails GNU Ethical Repository Criteria Evaluations

### ISSUE THREAD

**Issue by ghost** [strugee/gplenforced.org#8]
Kudos on setting up gplenforced.org :)

I understand that gplenforced.org is not a GNU project. The [GNU Ethical Repository Criteria Evaluations](https://www.fsf.org/news/gnu-releases-ethical-evaluations-of-code-hosting-services) are worth applying to non-GNU free software projects, as well as GNU ones.

gplenforced.org currently hosts its public code repository, wiki, and bug tracker with GitHub, which fails those criteria and also has [serious licensing concerns](https://joeyh.name/blog/entry/removing_everything_from_github/).

Please could gplenforced.org move its repo, wiki, and bug tracker to a host that does not suffer from those problems? The [github-backup](https://github-backup.branchable.com/) tool may help :)

As for which host to choose that is free from those problems, options include:
- [GitLab](https://gitlab.com)
- [NotABug](https://notabug.org)
- [GNU Savannah](https://savannah.nongnu.org/)

Thanks!

**Comment by strugee** [comment]
> Kudos on setting up gplenforced.org :)

Thanks :)

> gplenforced.org currently hosts its public code repository, wiki, and bug tracker with GitHub, which fails those criteria and also has serious licensing concerns.

Thanks for bringing this up - I probably should've done a better job addressing this concern.

I'm aware of the GNU guidelines. The problem here is that the target audience for gplenforced.org isn't really people who are already into free software - it's people who might have heard of it but haven't looked into it a lot, know a bit about it but don't e.g. know what GPL enforcement is, or about GNU (including the criteria that you just referenced). Most of these people will be on GitHub, and I wanted to meet them where they are. However, I understand that this is generally problematic, which is why [I take patches and bug reports via email][1] (though probably I could've done a better job communicating that reasoning).

Because of that I'm still leaning towards keeping the project on GitHub. But I'll leave this open in case anyone has thoughts. (FWIW if we ever did switch, we'd switch to GitLab.)

 [1]: https://github.com/strugee/gplenforced.org#contributing-changes

**Comment by strugee** [comment]
I also want to point out that re: licensing concerns, my understanding is that that was resolved and GitHub's intention was never to ban GPL'd works from GitHub or anything like that. Agreed that they could have handled it _way_ better but AFAIK that's a non-issue these days.

Had they actually made it impossible to upload GPL'd works to GitHub without watering down the GPL's terms, gplenforced.org would be off GitHub the instant that was confirmed.

**Comment by ghost** [comment]
@strugee wrote:
> the target audience for gplenforced.org isn't really people who are already into free software

I had to think about that for a second ;)

I think what you're getting at is that there are really *two* target audiences for gplenforced.org:

1. **project owners**: people who:
    - have a project that uses the GPL; *and*
    - believe in the Community-Oriented GPL Enforcement Principles; *and*
    - intend to follow those principles in respect of their project; *and*
    - want a convenient way to state the above.
2. **project users**: people who:
    - read a project's website or README, and might click on the gplenforced badge if they don't know what it means.

Clearly, the first group is composed of "people who are already into free software". So, in your statement above, you must have been referring to the second group. 

I appreciate you might care about using the same code hosting service as people in the first group, but in that case GitLab.com might be a *better* choice for gplenforced.

And for the second group, it doesn't matter whether the gplenforced code is hosted at GitHub.com or GitLab.com, so might as well be the latter :)

> re: licensing concerns, my understanding is that that was resolved and GitHub's intention was never to ban GPL'd works from GitHub or anything like that

As far as I know, the ToS were not modified in response to the concerns :(

**Comment by strugee** [comment]
> I had to think about that for a second ;)

Hah, fair. Sorry for the confusion :P

I think what I'm trying to get at is the idea that audience 2 (in your comment above) may transition into audience 1. E.g. a project maintainer (who knows about open source but not really about free software - seems pretty common, anecdotally) is looking around GitHub, sees the "GPL enforced" badge, and clicks on it. They then learn about enforcement and decide they believe in it too, so they add the badge to their READMEs and potentially check out the project source. If they check out this project it'd be nice for it to be on GitHub. Though I dunno, there isn't a lot of work to be done here; it's mostly a static site... not sure if this would really happen that often.

I'll think about it some more.

> As far as I know, the ToS were not modified in response to the concerns :(

https://www.fsf.org/blogs/licensing/do-githubs-updated-terms-of-service-conflict-with-copyleft

**Comment by ghost** [comment]
> I think what I'm trying to get at is the idea that audience 2 (in your comment above) may transition into audience 1

Yep, I understand that. My point is that this doesn't in any way rely on hosting gplenforced.org's code on GitHub ;)

> E.g. a project maintainer (who knows about open source but not really about free software - seems pretty common, anecdotally) is looking around GitHub, sees the "GPL enforced" badge, and clicks on it. They then learn about enforcement and decide they believe in it too, so they add the badge to their READMEs

This part does not rely on hosting gplenforced.org's code on GitHub, agreed? :)

> and potentially check out the project source. If they check out this project it'd be nice for it to be on GitHub.

Would it? ;) Mightn't it be even nicer if it was on GitLab, with a short, prominent FAQ a bit like this:

> ### *Why are you using GitLab instead of GitHub to store your code?*
> 
> *Because GitLab is [free software](https://en.wikipedia.org/w/index.php?title=Free_software&oldid=744934370#Definition_and_the_Four_Freedoms), whereas GitHub is not.*
> 
> ### *Can I still embed the GPLEnforced badge in my README at GitHub?*
> 
> *Please do!*

That way, an interested visitor doesn't just learn about GPL enforcement, they also learn about a free software alternative to GitHub :)

**Comment by johnjago** [comment]
I agree that moving the project to GitLab will help "practice what you preach", and it is a much easier task when the project is small, as it currently is.

**Comment by strugee** [comment]
Yeah, I think we'll move. Not sure when I'll have time though; some stuff in my personal life has taken up a lot of my time recently.

**Comment by ghost** [comment]
@strugee, all good wishes for the personal stuff.

I agree with @johnjago that project moves are easier while the number of issues/webhooks/wikis/etc is low. I also know that wellbeing is generally a prerequisite for good work, and it's totally understandable that your personal life should take priority. Kudos again on getting gplenforced.org off the ground, and thumbs up for the move in due course, when you have time to devote to it again :)
